network/network.go

```diff
@@ -0,0 +1,68 @@
+package network
+
+import (
+	"golang.org/x/net/context"
+
+	"github.com/dedis/cothority/monitor"
+)
+
+// Conn is the basic interface to represent any communication mean
+// between two host.
+type Conn interface {
+	// Send a message through the connection.
+	// obj should be a POINTER to the actual struct to send, or an interface.
+	// It should not be a Golang type.
+	Send(ctx context.Context, obj Body) error
+	// Receive any message through the connection. It is a blocking call that
+	// returns either when a message arrived or when Close() has been called, or
+	// when a network error occured.
+	Receive(ctx context.Context) (Packet, error)
+	// Close will close the connection. Implementations must take care that
+	// Close() makes Receive() returns with an error, and any subsequent Send()
+	// will return with an error. Calling Close() on a closed Conn will return
+	// ErrClosed.
+	Close() error
+
+	// Type returns the type of this connection
+	Type() ConnType
+	// Gives the address of the remote endpoint
+	Remote() Address
+	// Returns the local address and port
+	Local() Address
+	// XXX Can we remove that ?
+	monitor.CounterIO
```

Comment:
I just put that as `Tx()` & `Rx()` methods so it still implements the monitor.CounterIO interface but not explicitely now. They're usefuls anyway.